pkg/apis/crd/v1alpha1/types.go

```diff
@@ -600,3 +600,36 @@ type NamespacedName struct {
 	Name      string `json:"name,omitempty"`
 	Namespace string `json:"namespace,omitempty"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ExternalNode struct {
```

Comment:
Could you add some comments here?

Reply:
Thanks Jianjun, I have added a description, could you help take another look?

Reply:
The current version works for me. Or you can add "bare-metal server" too.